File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.cpp
#include <algorithm>
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int count = 1;
        int min_value = nums[0];
        for (int i : nums) {
            if (i - min_value > k) {
                count++;
                min_value = i;
            }
        }
        return count;
    }
};

File: 2294-partition-array-such-that-maximum-difference-is-k/2294-partition-array-such-that-maximum-difference-is-k.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        count=1
        min_value=nums[0]
        for i in nums:
            if i-min_value>k:
                count+=1
                min_value=i
        return count    


        